analyzers/src/SonarAnalyzer.CSharp/Rules/SymbolicExecution/SymbolicExecutionRunner.cs

```diff
@@ -41,29 +42,40 @@
 namespace SonarAnalyzer.Rules.SymbolicExecution
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    [Rule(ConditionEvaluatesToConstant.S2583DiagnosticId, LanguageNames.CSharp)]
+    [Rule(ConditionEvaluatesToConstant.S2589DiagnosticId, LanguageNames.CSharp)]
+    [Rule(EmptyCollectionsShouldNotBeEnumerated.DiagnosticId, LanguageNames.CSharp)]
+    [Rule(EmptyNullableValueAccess.DiagnosticId, LanguageNames.CSharp)]
+    [Rule(InitializationVectorShouldBeRandom.DiagnosticId, LanguageNames.CSharp)]
+    [Rule(InvalidCastToInterfaceSymbolicExecution.DiagnosticId, LanguageNames.CSharp)]
     [Rule(LocksReleasedAllPathsBase.DiagnosticId, LanguageNames.CSharp)]
+    [Rule(HashesShouldHaveUnpredictableSalt.DiagnosticId, LanguageNames.CSharp)]
+    [Rule(NullPointerDereference.DiagnosticId, LanguageNames.CSharp)]
+    [Rule(ObjectsShouldNotBeDisposedMoreThanOnce.DiagnosticId, LanguageNames.CSharp)]
+    [Rule(PublicMethodArgumentsShouldBeCheckedForNull.DiagnosticId, LanguageNames.CSharp)]
+    [Rule(RestrictDeserializedTypes.DiagnosticId, LanguageNames.CSharp)]
     public sealed partial class SymbolicExecutionRunner : SonarDiagnosticAnalyzer
     {
         private static readonly ImmutableDictionary<DiagnosticDescriptor, RuleFactory> AllRules = ImmutableDictionary<DiagnosticDescriptor, RuleFactory>.Empty
             .Add(LocksReleasedAllPaths.S2222, CreateFactory<LocksReleasedAllPaths>());
-        private readonly SymbolicExecutionAnalyzerFactory analyzerFactory;  // ToDo: This should be eventually removed
+        private static readonly ImmutableArray<ISymbolicExecutionAnalyzer> SonarRules = ImmutableArray.Create<ISymbolicExecutionAnalyzer>(    // ToDo: This should be migrated to AllRules
```

Comment:
Can't we address this TODO in the current PR?

Reply:
No, it's too big refactoring :( I wanted this PR to be as atomic as possible. I'll definitely migrate those later.